lib/client.js

```diff
@@ -40,7 +42,12 @@ export class Client {
     }, headers);
 
     const requestArgs = { method, url, body, params, headers };
-    const request = this.connection.send(requestArgs); // todo: how is connection set?
+    const request = this[$request](requestArgs);
     return new Response(request, this.Promise); // todo: make Promise be something
   }
 }
+
+export function assertFunction(fn) {
+  const type = typeof fn;
+  if (type !== 'function') throw new Error(`Expecting function, got ${type}`);
```

Comment:
Not sure how explicit you want to be with errors but this could throw a `TypeError`